test/files/run/t12292.check

```diff
@@ -0,0 +1,30 @@
+
+scala> import scala.annotation.nowarn
+import scala.annotation.nowarn
+
+scala> scala.#::.unapply(Stream(1))
+                 ^
+       warning: method unapply in object #:: is deprecated (since 2.13.0): Prefer LazyList instead
+                         ^
+       warning: value Stream in package scala is deprecated (since 2.13.0): Use LazyList instead of Stream
+val res0: Option[(Int, Stream[Int])] = Some((1,Stream()))
+
+scala> scala.#::.unapply(Stream(1)): @nowarn
+val res1: Option[(Int, Stream[Int])] @scala.annotation.nowarn = Some((1,Stream()))
+
+scala> (scala.#::.unapply(Stream(1)): @nowarn)
+val res2: Option[(Int, Stream[Int])] @scala.annotation.nowarn = Some((1,Stream()))
+
+scala> scala.#::.unapply(Stream(1)): @inline
+                 ^
+       warning: method unapply in object #:: is deprecated (since 2.13.0): Prefer LazyList instead
+                         ^
+       warning: value Stream in package scala is deprecated (since 2.13.0): Use LazyList instead of Stream
+                                      ^
+       warning: type Stream in package scala is deprecated (since 2.13.0): Use LazyList instead of Stream
```

Comment:
why does this warning not occur for `res0`?

Reply:
Looking at the arrow, it's pointing at `@inline`.  All that "it's not actually a type but using type ascription syntax" Seth was talking about... looks like it's causing this type deprecation to be emitted I think.